src/Two/GithubProvider.php

```diff
@@ -32,13 +32,26 @@ protected function getTokenUrl()
 	 */
 	protected function getUserByToken($token)
 	{
-		$response = $this->getHttpClient()->get('https://api.github.com/user?access_token='.$token, [
+		$options = [
 			'headers' => [
 				'Accept' => 'application/vnd.github.v3+json',
 			],
-		]);
+		];
+
+		$response = $this->getHttpClient()->get('https://api.github.com/user?access_token='.$token, $options);
+		$user = json_decode($response->getBody(), true);
+
+		if (in_array('user:email', $this->scopes)) {
+			$response = $this->getHttpClient()->get('https://api.github.com/user/emails?access_token='.$token, $options);
+			$emails = json_decode($response->getBody(), true);
+			foreach ($emails as $email) {
+				if ($email['primary'] && $email['verified']) {
+					$user['email'] = $email['email'];
```

Comment:
Shouldn't we break out of the loop once we've added an email address? Minor optimization. :)